core/Services/Calendar/EventStore.vala

```diff
@@ -79,6 +87,29 @@ public class Calendar.EventStore : Object {
         this.month_start = Calendar.Util.datetime_get_start_of_month (get_page ());
         compute_ranges ();
 
+#if E_CAL_2_0
+        this.system_timezone = ECal.util_get_system_timezone ();
+#else
+        this.system_timezone = ECal.Util.get_system_timezone ();
+#endif
+        // Listen for system timezone changes
+        try {
+            SettingsManager settings_manager = Bus.get_proxy_sync (BusType.SYSTEM,
+                "org.freedesktop.timedate1", "/org/freedesktop/timedate1");
+            ((DBusProxy)settings_manager).g_properties_changed.connect ((changed, invalid) => {
```

Comment:
This does not seem to work (?). The signal handler is not triggered by changing the system timezone using `timedatectl` at the command line or otherwise (even though the wingpanel indicator time does change).  Not sure why or how to fix it :disappointed:

Reply:
Thanks for your suggestions. I'll have a look at this and see what I can figure out.